File: eventbridge-pipes-sqs-to-step-functions-cdk-dotnet/cdk/src/Cdk/CdkStack.cs
using System.Collections.Generic;

using Amazon.CDK;
using Amazon.CDK.AWS.IAM;
using Amazon.CDK.AWS.Pipes;
using Amazon.CDK.AWS.SQS;
using Amazon.CDK.AWS.StepFunctions;

using Constructs;

namespace Cdk
{
    public class CdkStack : Stack
    {
        internal CdkStack(
            Construct scope,
            string id,
            IStackProps props = null) : base(
            scope,
            id,
            props)
        {
            var source = new Queue(
                this,
                "SourceSQSQueue");

            var mapState = new Map(
                this,
                "LoopMessages",
                new MapProps()
                {
                    ItemsPath = JsonPath.EntirePayload,
                });
            mapState.Iterator(
                new Wait(
                    this,
                    "WaitState",
                    new WaitProps()
                    {
                        Time = WaitTime.Duration(Duration.Seconds(10)),
                    }));

            var stepFunction = new StateMachine(
                this,
                "TargetStepFunction",
                new StateMachineProps()
                {
                    Definition = mapState,
                    StateMachineType = StateMachineType.STANDARD
                });

            var sourcePolicy = new PolicyDocument(
                new PolicyDocumentProps
                {
                    Statements = new[]
                    {
                        new PolicyStatement(
                            new PolicyStatementProps
                            {
                                Resources = new[] { source.QueueArn },
                                Actions = new[] { "sqs:ReceiveMessage", "sqs:DeleteMessage", "sqs:GetQueueAttributes" },
                                Effect = Effect.ALLOW
                            })
                    }
                });

            var targetPolicy = new PolicyDocument(
                new PolicyDocumentProps
                {
                    Statements = new[]
                    {
                        new PolicyStatement(
                            new PolicyStatementProps
                            {
                                Resources = new[] { stepFunction.StateMachineArn },
                                Actions = new[] { "states:StartExecution" },
                                Effect = Effect.ALLOW
                            })
                    }
                });

            var pipeRole = new Role(
                this,
                "PipeRole",
                new RoleProps
                {
                    AssumedBy = new ServicePrincipal("pipes.amazonaws.com"),
                    InlinePolicies = new Dictionary<string, PolicyDocument>(2)
                    {
                        { "SourcePolicy", sourcePolicy },
                        { "TargetPolicy", targetPolicy }
                    }
                });

            var pipe = new CfnPipe(
                this,
                "Pipe",
                new CfnPipeProps
                {
                    RoleArn = pipeRole.RoleArn,
                    Source = source.QueueArn,
                    SourceParameters = new CfnPipe.PipeSourceParametersProperty()
                    {
                        SqsQueueParameters = new CfnPipe.PipeSourceSqsQueueParametersProperty
                        {
                            BatchSize = 5,
                            MaximumBatchingWindowInSeconds = 120
                        }
                    },
                    Target = stepFunction.StateMachineArn,
                    TargetParameters = new CfnPipe.PipeTargetParametersProperty
                    {
                        StepFunctionStateMachineParameters = new CfnPipe.PipeTargetStateMachineParametersProperty()
                        {
                            InvocationType = "FIRE_AND_FORGET"
                        }
                    }
                });

            var outputQueueName = new CfnOutput(
                this,
                "QueueUrlOutput",
                new CfnOutputProps()
                {
                    ExportName = "QueueUrlOutput",
                    Value = source.QueueName
                });
        }
    }
}

File: eventbridge-pipes-sqs-to-stepfunctions-filter-transform-python-cdk/src/cdkstack.py
from aws_cdk import (
    CfnOutput,
    Stack,
    aws_iam as iam,
    aws_pipes as pipes,
    aws_sqs as sqs,
    aws_stepfunctions as sfn,
)
from constructs import Construct
import json

class SqsToStepfunctionsFilterTransformStack(Stack):

    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        sourcepipe = sqs.Queue(self, 'sqsqueue-sourcepipe')
        

        targetpipe = sfn.StateMachine(self, "sf-targetqueue-state-machine",
            definition=sfn.Pass(self, "start-state")
        )

        source_policy = iam.PolicyStatement(
                actions=['sqs:ReceiveMessage', 'sqs:DeleteMessage', 'sqs:GetQueueAttributes'],
                resources=[sourcepipe.queue_arn],
                effect=iam.Effect.ALLOW,
        )

        target_policy = iam.PolicyStatement(
                actions=['states:StartExecution'],
                resources=[targetpipe.state_machine_arn],
                effect=iam.Effect.ALLOW,
        )

        pipe_role = iam.Role(self, 'pipe-role',
            assumed_by=iam.ServicePrincipal('pipes.amazonaws.com'),
        )

        pipe_role.add_to_policy(source_policy)
        pipe_role.add_to_policy(target_policy)

        pipe = pipes.CfnPipe(self, "pipe",
            role_arn=pipe_role.role_arn,
            source=sourcepipe.queue_arn,
            source_parameters=pipes.CfnPipe.PipeSourceParametersProperty(
                sqs_queue_parameters=pipes.CfnPipe.PipeSourceSqsQueueParametersProperty(
                    batch_size=1
                )
            
            ,
            
            filter_criteria=pipes.CfnPipe.FilterCriteriaProperty(
                filters=[pipes.CfnPipe.FilterProperty(
                    pattern=json.dumps({"body":{"status": ["COMPLETE"]}})
                    )]
                    )
            
            
            ),
            
            target=targetpipe.state_machine_arn,
            target_parameters=pipes.CfnPipe.PipeTargetParametersProperty(
                step_function_state_machine_parameters=pipes.CfnPipe.PipeTargetStateMachineParametersProperty(
                    invocation_type="FIRE_AND_FORGET"
                ),
                input_template=json.dumps({
                    "playerid": "<$.body.id>",
                    "teamname": "<$.body.team>",
                    "teamstatus": "<$.body.status>"
                })
            )
        )


        # Output
        CfnOutput(self, "SQS_QUEUE_URL", value=sourcepipe.queue_url)
        CfnOutput(self, "EVENTBRIDGE_PIPE_ARN", value=pipe.attr_arn)
        CfnOutput(self, "STATE_MACHINE_ARN", value=targetpipe.state_machine_arn)
